File: floatmenu/src/main/java/com/gurunars/floatmenu/AnimationListener.java
package com.gurunars.floatmenu;

/**
 * Handler for FAB open/close animations.
 */
public interface AnimationListener {

    /**
     * Called when the animation is initiated.
     *
     * @param projectedDuration time in millis that theoretically should
     */
    void onStart(int projectedDuration);

    /**
     * Called when the animation is completed.
     */
    void onFinish();

    class Default implements AnimationListener {
        @Override
        public void onStart(int projectedDuration) {

        }

        @Override
        public void onFinish() {

        }
    }

}

File: floatmenu/src/main/java/com/gurunars/floatmenu/AnimationListener.kt
package com.gurunars.floatmenu

/**
 * Handler for FAB open/close animations.
 */
interface AnimationListener {

    /**
     * Called when the animation is initiated.
     *
     * @param projectedDuration time in millis that theoretically should
     */
    fun onStart(projectedDuration: Int)

    /**
     * Called when the animation is completed.
     */
    fun onFinish()

    class Default : AnimationListener {
        override fun onStart(projectedDuration: Int) {

        }

        override fun onFinish() {

        }
    }

}
